drivers/vmwarevsphere/cloudinit.go

```diff
@@ -227,31 +227,50 @@ func (d *Driver) mountCloudInitIso(vm *object.VirtualMachine, dc *object.Datacen
 	return vm.AddDevice(d.getCtx(), add...)
 }
 
-func addSSHUserToYaml(yamlcontent, user, sshkey string) (string, error) {
+func addSSHUserToYaml(yamlcontent, user, group, sshkey string) (string, error) {
 	cf := make(map[interface{}]interface{})
 	if err := yaml.Unmarshal([]byte(yamlcontent), &cf); err != nil {
 		return "", err
 	}
 
 	newUser := map[interface{}]interface{}{
-		"name":        user,
-		"sudo":        "ALL=(ALL) NOPASSWD:ALL",
-		"lock_passwd": "true",
+		"name":          user,
+		"sudo":          "ALL=(ALL) NOPASSWD:ALL",
+		"lock_passwd":   true,
+		"groups":        group,
+		"create_groups": false,
+		"no_user_group": true,
 		"ssh_authorized_keys": []string{
 			sshkey,
 		},
 	}
 
 	if val, ok := cf["users"]; ok {
 		u := val.([]interface{})
-		u = append(u, newUser)
-		cf["users"] = u
+		cf["users"] = append(u, newUser)
 	} else {
 		users := make([]interface{}, 1)
 		users[0] = newUser
 		cf["users"] = users
 	}
 
+	if val, ok := cf["groups"]; ok {
+		g := val.([]interface{})
+		var exists = false
+		for _, v := range g {
+			if v.(string) == group {
```

Comment:
may want to safely convert v, with something like:
`if something, _ := v.(string); something == group {`

Reply:
great suggestion